File: parciales/ED1-Parcial1-2021-1--001/codigo/Recursion.java
public class Recursion{
  private static int solve(int[] a, int i, int j){
    if (i == j) return a[i];
    return Math.max(a[i] - solve(a, i + 1, j), a[j] - solve(a, i, j - 1));
  }
  public static int solve(int[] a){
    return solve(a, 0, a.length -1); 
  }
  public static void main(String[] args){
     System.out.println(solve(new int[]{10, 80, 90, 30}));
     System.out.println(solve(new int[]{10, 20, 30, 50}));
  }
}

File: parciales/ED1-Parcial1-2021-1--001/codigo/Recursion.py
def solveAUX(a, i, j):
    if i == j: 
      return a[i]
    return max(a[i] - solveAUX(a, i + 1, j), a[j] - solveAUX(a, i, j - 1))
  
def solve(a):
    return solveAUX(a, 0, len(a) - 1)

def __main__():
    print(solve([10, 80, 90, 30]))
    print(solve([10, 20, 30, 50]))

__main__()
  
